src/Combinatorics/EnumerativeCombinatorics/multipartitions.jl

```diff
@@ -0,0 +1,166 @@
+################################################################################
+# Multipartitions.
+#
+# Copyright (C) 2020 Ulrich Thiel, ulthiel.com/math
+#
+# Originally taken from the JuLie [repository](https://github.com/ulthiel/JuLie)
+# by Ulrich Thiel and OSCAR-ified by Morgan Rodgers.
+################################################################################
+
+
+
+
+"""
+    multipartition(mp::Vector{Partition{T}}) where T <: IntegerUnion
+    multipartition(mp::Vector{Vector{T}}) where T <: IntegerUnion
+
+Return the multipartition given by the vector `mp` of integer sequences `mp`
+(which are interpreted as integer partitions) as an object of type
+`Multipartition{T}`.
+
+The element type `T` may be optionally specified, see also the examples below.
+
+# Examples
+```jldoctest
+julia> P = multipartition([[2,1], [], [3,2,1]])
+Partition{Int64}[[2, 1], [], [3, 2, 1]]
+julia> sum(P)
+9
+julia> P[2]
+Empty partition
+julia> P = multipartition(Vector{Int8}[[2,1], [], [3,2,1]])
+Partition{Int8}[[2, 1], [], [3, 2, 1]]
+```
+
+# References
+1. Wikipedia, [Multipartition](https://en.wikipedia.org/wiki/Multipartition)
+"""
+function multipartition(mp::Vector{Partition{T}}) where T <: IntegerUnion
+  return Multipartition(mp)
+end
+
+function multipartition(mp::Vector{Vector{T}}) where T <: IntegerUnion
+  return Multipartition([partition(p) for p in mp])
+end
+
+# This is only called when the empty array is part of mp (because then it's
+# "Any" type and not of Integer type).
+function multipartition(mp::Vector{Vector{Any}})
+  return Multipartition([partition(p) for p in mp])
+end
+
+function Base.show(io::IO, ::MIME"text/plain", MP::Multipartition)
+  print(io, MP.mp)
+end
+
+function Base.size(MP::Multipartition)
+  return size(MP.mp)
+end
+
+function Base.length(MP::Multipartition)
+  return length(MP.mp)
+end
+
+function Base.getindex(MP::Multipartition, i::Int)
+  return getindex(MP.mp,i)
+end
+
+
+"""
+    sum(P::Multipartition{T}) where T<:IntegerUnion
+
+If `P` is a multipartition of the integer n, this function returns n.
+"""
+function Base.sum(MP::Multipartition{T}) where T<:IntegerUnion
+  return sum(sum, MP.mp)
```

Comment:
```suggestion
  return sum(sum, MP.mp; init=zero(T))
```
to fix `sum(multipartition(Vector{Int}[]))`